src/Shared/Contracts/IManagerProvider.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+
+namespace Microsoft.CST.OpenSource.Contracts;
+
+using PackageUrl;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+/// <summary>
+/// The interface to implement project manager specific methods.
+/// Useful for mocking when doing unit tests.
+/// </summary>
+public interface IManagerProvider
```

Comment:
Nit: the name `IManagerProvider` sounds to me like something that can provide Managers (e.g. a manager factory). That doesn't really seem like what implementations of this type is doing - it seems like the methods here are more focused on particular package actions? 

I wonder if there could be a better name that conveys that more clearly - as a brainstorm, would something like `IManagerPackageActions` make sense?

Reply:
@brbayes-msft and I were chatting about this. We didn't have a better name at the time. I like your suggestion.
Any thoughts @gfs ?

Reply:
I renamed it to `IManagerPackageActions`, we can change it pretty easily if we decide otherwise. But it makes a lot more sense than `IManagerProvider`